processing/1_create_compendia.py

```diff
@@ -0,0 +1,255 @@
+# We will need to remove these so that the count calculations are accurate
+sample_to_strain_table_full_processed = sample_to_strain_table_full[
+    ~sample_to_strain_table_full.index.duplicated(keep="first")
+]
+
+assert (
+    len(sample_to_strain_table_full.index.unique())
+    == sample_to_strain_table_full_processed.shape[0]
+)
+
+# +
+# Aggregate boolean labels into a single strain label
+aggregated_label = []
+for exp_id in list(sample_to_strain_table_full_processed.index):
+    if sample_to_strain_table_full_processed.loc[exp_id, "PAO1"].all() == True:
+        aggregated_label.append("PAO1")
+    elif sample_to_strain_table_full_processed.loc[exp_id, "PA14"].all() == True:
+        aggregated_label.append("PA14")
+    elif sample_to_strain_table_full_processed.loc[exp_id, "PAK"].all() == True:
+        aggregated_label.append("PAK")
+    elif (
+        sample_to_strain_table_full_processed.loc[exp_id, "ClinicalIsolate"].all()
+        == True
+    ):
+        aggregated_label.append("Clinical Isolate")
+    else:
+        aggregated_label.append("NA")
+
+sample_to_strain_table_full_processed["Strain type"] = aggregated_label
+
+sample_to_strain_table = sample_to_strain_table_full_processed["Strain type"].to_frame()
+
+sample_to_strain_table.head()
+# -
+
+# ## Bin samples as PAO1 or PA14
+#
+# * Bin samples based on threshold from previous notebook
+# * Check if there are any samples that have a high mapping to both PAO1 and PA14 (i.e. ambiguous mapping)
+
+# Add column calculating the difference in mapping rates
+pao1_logs["diff_mapping_rate"] = pao1_logs["mapping_rate"] - pa14_logs["mapping_rate"]
+pa14_logs["diff_mapping_rate"] = pa14_logs["mapping_rate"] - pao1_logs["mapping_rate"]
+
+# +
+high_pao1_mapping_ids = list(
+    pao1_logs.query(
+        "mapping_rate>=@mapping_threshold&diff_mapping_rate>@diff_mapping_threshold"
+    ).index
+)
+high_pa14_mapping_ids = list(
+    pa14_logs.query(
+        "mapping_rate>=@mapping_threshold&diff_mapping_rate>@diff_mapping_threshold"
+    ).index
+)
+
+"""high_pao1_mapping_ids = list(
+    pao1_logs.query("mapping_rate>=@mapping_threshold&diff_mapping_rate>@diff_mapping_threshold_min&diff_mapping_rate<@diff_mapping_threshold_max").index
+)
+high_pa14_mapping_ids = list(
+    pa14_logs.query("mapping_rate>=@mapping_threshold&diff_mapping_rate>@diff_mapping_threshold_min&diff_mapping_rate<@diff_mapping_threshold_max").index
+)"""
+
+print(len(high_pao1_mapping_ids))
+print(len(high_pa14_mapping_ids))
+
+# +
+# Check if any ids have high mapping rate for both PAO1 and PA14
+high_pao1_pa14_mapping_ids = list(
+    set(high_pao1_mapping_ids).intersection(high_pa14_mapping_ids)
+)
+
+print(len(high_pao1_pa14_mapping_ids))
+# -
+
+# **Some observations:**
+# * Looks like there are not any samples that map to both PAO1 and PA14 using our criteria
+# * The number of PA14 samples is much lower compared to PAO1. Does this mean that the mapping rates of PA14 samples mapped to PA14 reference lower?
+
```

Comment:
That's an interesting point. Looking at the mapping rate plot in `0_decide_thresholds`, it looks like the variance of the second hump in PAO1 and PA14 are slightly different, so the cutoff of 0.3 may be too strict for PA14? This may be overkill, but since you have the labels for some of the strains could you plot the mapping rates of those samples only so you can get a sense of the  "true" mapping rate distributions and then see what proportion of samples are outside of 30% mapping rate so you can maybe get an estimate of the FDR for that cutoff?

Reply:
This is a great comment! I think that looking at this as well as the next point you mentioned will help me to determine if mapping rate is an appropriate metric to use to bin samples.